src/app/components/dynamic-table/dynamic-table.component.ts

```diff
@@ -72,7 +77,6 @@ export class DynamicTableComponent
         this.dataSource.data = [];
         this.displayedColumns = Object.keys(data[0]);
 
-        /* eslint-disable  @typescript-eslint/no-explicit-any */
         const dataSource = this.dataSource.data;
         data.forEach((field: any) => {
```

Comment:
This should be a union of single schema field or single data object. I think you should create a type alias for this.